src/cli/hook-utils/get-check-update-data.js

```diff
@@ -0,0 +1,334 @@
+  const jsonFiles = ['package.json'];
+  const jsonDepFiles = [];
+  const inaccessibleFiles = [];
+
+  for (const fileName of jsonFiles) {
+    try {
+      const jsonFile = await getJSON(`${cwd}/${fileName}`);
+      const jsonIsParsable =
+        jsonFile &&
+        typeof jsonFile === 'object' &&
+        !Array.isArray(jsonFile) &&
+        jsonFile.dependencies;
+
+      if (jsonIsParsable) {
+        jsonDepFiles.push(fileName);
+      }
+    } catch (err) {
+      inaccessibleFiles.push({ name: fileName, error: err });
+    }
+  }
+
+  return { jsonDepFiles, inaccessibleFiles };
+}
+
+/**
+ * Pulls dependencies from a given file and JSON structure.
+ * @param json JSON information that includes dependencies
+ * @param fileName name of the file that the dependency list is coming from
+ */
+dependencyExports.extractDependencies = async (json, fileName) => {
+  // Determine if the JSON passed is an object
+  const jsonIsParsable =
+    json !== null && typeof json === 'object' && !Array.isArray(json);
+
+  if (jsonIsParsable) {
+    let boltCurrentVersion = '';
+    let denoCurrentVersion = '';
+    if (json['dependencies']['@slack/bolt']) {
+      const boltCurrentVersionOutput = JSON.parse(
+        await dependencyExports.getBoltCurrentVersion()
+      );
+
+      if (boltCurrentVersionOutput !== '') {
+        boltCurrentVersion =
+          boltCurrentVersionOutput['dependencies']['@slack/bolt']['version'];
+      }
+    }
+    if (json['dependencies']['@slack/deno-slack-sdk']) {
+      const denoCurrentVersionOutput = JSON.parse(
+        await dependencyExports.getDenoCurrentVersion()
+      );
+
+      if (denoCurrentVersionOutput !== '') {
+        denoCurrentVersion =
+          denoCurrentVersionOutput['dependencies']['@slack/deno-slack-sdk'][
+            'version'
+          ];
+      }
+    }
+
+    return [
+      [
+        '@slack/bolt',
+        {
+          version: boltCurrentVersion,
+        },
+      ],
+      [
+        '@slack/deno-slack-sdk',
+        {
+          version: denoCurrentVersion,
+        },
+      ],
+    ];
+  }
+
+  return [];
+};
+
+/**
+ * Gets the latest module version.
+ * @param moduleName the module that the latest version is being queried for
+ */
+async function fetchLatestModuleVersion(moduleName) {
+  let command = "";
+  if (moduleName === '@slack/bolt') {
+    command = `npm info ${moduleName} version --tag next-gen`;
+  } else if (moduleName === '@slack/deno-slack-sdk') {
+    command = `npm info ${moduleName} version --tag latest`;
+  }
+  const { stdout } = await exec(command).catch((err) => {
+    throw new Error(err);
+  });
+
+  return stdout ? stdout.replace(/\n$/, '') : '';
+}
+
+/**
+ * Checks if a dependency's upgrade from a current to the latest version will cause a
+ * breaking change.
+ * @param current current dependency version in project
+ * @param latest most up-to-date dependency version available on NPM
+ */
+function hasBreakingChange(current, latest) {
```

Comment:
Should we add / do we already have messaging indicating a breaking change on the CLI side? I didn't notice any when testing this, but may have overlooked it!

Reply:
ahh, I see! I saw that the breaking change specific messaging surfaced in the `slack update` command and was wondering if we should also surface it in `slack doctor`, but also I think as long as it's surfaced somewhere then that works!! 👍 thanks for pointing that out!